js/client.js

```diff
@@ -1636,7 +1651,19 @@
 			document.title = room.title ? room.title + " - Showdown!" : "Showdown!";
 		},
 		updateAutojoin: function () {
-			if (Config.server.id !== 'showdown') return;
+			if (!Config.server.registered) return;
+			var curAutojoin = (Tools.prefs('autojoin'));
+			var useString = false;
```

Comment:
If you default `useString` to `true`, you can cut a few more lines. But considering you only read it once, you might want to remove it entirely and just use `typeof curAutojoin === 'string'`.